Libs/Mesh/Mesh.cpp

```diff
@@ -696,6 +696,28 @@ Mesh& Mesh::setField(std::string name, Array array)
   return *this;
 }
 
+Mesh& Mesh::setFieldForFaces(std::string name, Array array)
+{
+  if (!array)
+    throw std::invalid_argument("Invalid array.");
+
+  if (name.empty())
+    throw std::invalid_argument("Provide name for the new field");
+
+  int numVertices = mesh->GetPoints()->GetNumberOfPoints();
+  if (array->GetNumberOfTuples() != numVertices) {
+    std::cerr << "WARNING: Added a mesh field with a different number of elements than points\n";
```

Comment:
This function is supposed to add a field for faces. So the number of faces needed to be considered